packages/frontend/src/fetcher/on-chain/index.ts

```diff
@@ -108,62 +112,70 @@ class Fetcher implements IPartialFetcher {
     }
 
     public async fetchAnswersHistory({
-        provider,
+        nodeClient,
         realityV3Address,
         questionId,
     }: FetchAnswersHistoryParams): Promise<RealityResponse[]> {
         if (!realityV3Address || !questionId) return [];
-        const { chainId } = await provider.getNetwork();
+        const chainId = await nodeClient.getChainId();
         enforce(
             chainId in SupportedChainId,
             `unsupported chain with id ${chainId}`
         );
-        const realityContract = new Contract(
-            realityV3Address,
-            REALITY_ETH_V3_ABI,
-            provider
-        );
 
-        let toBlock = await provider.getBlockNumber();
+        let toBlock = await nodeClient.getBlockNumber();
         let fromBlock = toBlock - LOGS_BLOCKS_SIZE;
         const answersFromLogs: RealityResponse[] = [];
         try {
             while (true) {
-                const logs = await provider.getLogs({
-                    address: realityContract.address,
+                // FIXME: check if this implementation still works properly
+                const newAnwersEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
+                    event: parseAbiItem(
+                        "event LogNewAnswer(bytes32,bytes32,bytes32,address,uint256,uint256,bool)"
+                    ),
                     fromBlock,
                     toBlock,
-                    topics: [
-                        [NEW_ANSWER_LOG_TOPIC, NEW_QUESTION_LOG_TOPIC],
-                        questionId,
-                    ],
                 });
+                const newQuestionEventLogs = await nodeClient.getLogs({
+                    address: realityV3Address as `0x${string}`,
```

Comment:
Ditto

Reply:
Fixed in f44ffd9afee046bbfeeca0ecb66ab8586fee4e83.